src/prototype.rs

```diff
@@ -144,7 +144,38 @@ impl<'a> FirstPass<'a> {
             return self.parse_indented_code_block(ix, remaining_space);
         }
 
+
+        // HTML Blocks
+
+        // HTML blocks may have 1-3 spaces in front of them, but our scanning
+        // functions don't know that. The spaces are preserved in the output,
+        // so we don't want to move `ix` yet.
         line_start.scan_all_space();
+        let nonspace_ix = start_ix + line_start.bytes_scanned();
+
+        // HTML block openings always happen on a single line, but our scanning
+        // functions are also used for inline HTML and may consume multiple lines,
+        // so we should only allow them to see one line at a time.
+        let line_end_ix = nonspace_ix + scan_line_ending(&self.text[nonspace_ix..]);
+        // Types 1-5 are all detected by one function and all end with the same
+        // pattern
+        if let Some(html_end_tag) = get_html_end_tag(&self.text[nonspace_ix..=line_end_ix]) {
```

Comment:
Why ..= here? Does it need to actually see the newline character? This could panic if the string ends without a newline (so `line_end_ix == text.len()`).